apollo-test/src/main/java/com/spotify/apollo/test/StubClient.java

```diff
@@ -275,13 +319,48 @@ private ResponseSource responseSource() {
 
   }
 
+
+  public class WhenResponseBuilder {
+
+    private final Matcher<Request> requestMatcher;
+
+    public WhenResponseBuilder(Matcher<Request> requestMatcher) {
+      this.requestMatcher = Objects.requireNonNull(requestMatcher);
+    }
+
+    /**
+     * Immediately respond with the supplied response to any matching request.
+     */
+    public void respond(Response<ByteString> response) {
+      addMatcherFirst(requestMatcher, Responses.constant(ResponseWithDelay.forResponse(response)));
+    }
+
+    /**
+     * Respond with responses and delays supplied by the provided ResponseSource to any matching
+     * request.
+     */
+    public void respond(ResponseSource responseSource) {
+      addMatcherFirst(requestMatcher, responseSource);
+    }
+  }
+
   @AutoValue
-  public static abstract class RequestResponsePair {
```

Comment:
Well, to be precise - I'm OK with breaking the API, since this is something that goes into 2.0, but this class still needs to be public, for people to be able to see what the StubClient has done.